src/utils/idpLoader.ts

```diff
@@ -29,24 +29,44 @@ export function parseIdpMetadata(
         }
         return "";
       });
-      const elementInfoOrErrors = IDPEntityDescriptor.decode({
-        cert: certs,
-        entityID: element.getAttribute("entityID"),
-        entryPoint: (element
-          .getElementsByTagName(SingleSignOnServiceTAG)
-          .item(0) as Element).getAttribute("Location"),
-        logoutUrl: (element
-          .getElementsByTagName(SingleLogoutServiceTAG)
-          .item(0) as Element).getAttribute("Location")
-      });
-      if (elementInfoOrErrors.isLeft()) {
+      try {
+        const elementInfoOrErrors = IDPEntityDescriptor.decode({
```

Comment:
nit: the pattern to name eithers / validation is `errorOrValue`. In this case it would be `errorOrElementInfo` (so 'error' stays on the 'left' side)